docs/installation/updating-with-npm.md

```diff
@@ -0,0 +1,39 @@
+# Updating GOV.UK Frontend with node package manager (NPM)
+
+You can update with node package manager (NPM) if you [originally installed
+GOV.UK Frontend with NPM](https://github.com/alphagov/govuk-frontend/blob/master/docs/installation/installing-with-npm.md#requirements).
+
+## Find out which version you're using
+
+You can find out which version of GOV.UK Frontend your project is using with:
+
+```shell
+npm list govuk-frontend
+```
+
+If you do not have command line access, you can see the version number in the
+`package.json` file in the root of your project directory. For example:
+
+```json
+'gov-uk frontend': ‘2.11.0’
+````
+
+## Update
+
+You can check the [release
+notes](https://github.com/alphagov/frontend/releases) in the GOV.UK Frontend Github repository to find out the latest version of GOV.UK Frontend.
+
+You can update to any version.
+
+You'll need to progressively update to each version following your current
+version. You will not need to make code changes for GOV.UK Frontend to keep working in
```

Comment:
Do we want to talk about semantic versioning here? It _might_ be easier for the user if they know that e.g. a 2.x to 2.y change will not include breaking changes, rather than requiring them to read the release notes to find that out?

Reply:
So something like: "You will not need to make code changes for GOV.UK Frontend to keep working in your project, unless you're updating to a major version - for example 2.0 rather than 2.13.0"?

Reply:
Could I propose for our style that we avoid 

'major', 'minor', 'patch'

and instead always use

'breaking', 'feature', 'fix'?

I think these are more understandable and reflect the CHANGELOG

Reply:
I like that idea a lot when we're talking about an individual release (e.g. 3.0.0 is a 'breaking release'), but I don't know how well it works when we're talking about how _version numbers_ change?

For example, someone updating from v2.13.0 to v3.1.1 is updating to a 'fix' release, but because the major version number changes they know there'll be breaking changes anyway.

We could talk about 'if the major version number changes' (having explained that term somewhere) but I'm not sure we'd ever say 'if the breaking version number changes'

Reply:
I think that's just a matter of how it's worded specifically what I was responding to:

> unless you're updating to a major version

would be

> unless you're updating to a version with breaking changes

etc

something to talk about later probs

Reply:
How about combining the two, so we clarify it for anyone who might not be clear on either term? Something like: "...unless you're updating to a major version. A version is major if it has a breaking changes section in its release notes."

Reply:
>"...unless you're updating to a major version. A version is major if it has a breaking changes section in its release notes."

It's slightly more nuanced than that though, because the specific version they're updating to is irrelevant unless they happen to be running the immediate preceding release. In the example I gave above, going from v2.13.0 to v3.1.1 is _not_ 'updating to a major version' (the version they're updating to is a patch release!) but the major version number does change.

I guess we need to work out what's easier for the user – going through the release notes to understand whether there was a breaking change _at some point_, or understanding the basics of how major, minor and patch version numbers work?

Reply:
How about this (to cover this discussion and Ollie's comment about 'progressive' updates too):

You will need to make code changes to make sure GOV.UK keeps working in your project, if you’re updating:

- to a major version - for example you’re updating to 2.0.0
- past a major version - for example you’re updating from 1.2.0 to 2.3.0

A version is major ('breaking') if there's a 'Breaking changes' section in the release notes.

Reply:
That sounds ace, thanks @m-green 👏 

I wonder if it should say 'You *may* need to make code changes' (sometimes, the 'breaking' part of the release might not affect a user) and whether we need to point the user back to the release notes so that they know what those code changes might be?